src/main/scala/com/mongodb/spark/sql/types/BsonCompatibility.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.spark.sql.types
+
+import scala.util.{Failure, Success, Try}
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
+import org.apache.spark.sql.types._
+
+import org.bson._
+import org.bson.types.{ObjectId => NativeObjectId}
+
+private[spark] object BsonCompatibility {
+
+  trait CompatibilityBase[T <: BsonValue] {
```

Comment:
Its just for use inside the `BsonCompatibility` essentially making it a sealed trait.